src/dev/impl/DevToys/UI/Controls/CodeEditor.xaml

```diff
@@ -126,6 +140,22 @@
                 DiffRightText="{x:Bind DiffRightText, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}"
                 AutomationProperties.LabeledBy="{x:Bind HeaderTextBlock}"/>-->
         </Grid>
+        <Border
```

Comment:
I think I start to understand better why you need the RunIdleAsync 🤔
If I follow correctly the logic, you have a Border that stretches so it fill-in its parent. Then you have this ContentControl that doesn't have the HorizontalAlignment and VerticalAlignment properties set to stretch. Therefore, to make it fitting, you listen to the SizeChanged event of the Border.

I admit I didn't give a try at the code yet, but seeing this make me wonder "why?". Why is this needed? Why not:
1. Setting VerticalAlignment and HorizontalAlignment to Stretch to the ContentControl?
2. Why not making the ContentControl a child of the Border?
```suggestion
        <Border
```

Reply:
I believe you are partly right, but let me clarify a few things.

> you have a Border that stretches so it fill-in its parent. Then you have this ContentControl that doesn't have the HorizontalAlignment and VerticalAlignment properties set to stretch. Therefore, to make it fitting, you listen to the SizeChanged event of the Border.

Yes, this is right. Actually, the Border technique is one that I stole from elsewhere in DevToys ([it is used for CustomTextBox](https://github.com/veler/DevToys/blame/main/src/dev/impl/DevToys/UI/Controls/CustomTextBox.xaml.cs#L856-L871)). I believe TextBox has the same problem as DataGrid which is that, without a fixed height, it will naturally use all of the vertical space, causing the parent page to scroll rather than the control itself. I think the Border technique is actually quite cool (by applying the sizing rules of one control to another).

> you need, sometime, to wait for the Border to get its size recalculated so you can give the ContentControl the correct size.

No, not quite. I think the Border technique should work fine (and does for CustomTextBux) But there is a problem which I cannot explain. When the DataGrid is displayed and the user collapses the expanded view, the Border sizing is wrong. It requests too much vertical space, despite not having any contents. This is where my second hacky trick comes into play. I briefly show the CodeEditor. Somehow, it sizes much better and fixes the Border. At that point I can put back the DataGrid, which is sized according to the Border.

Here is a video of the problem.

https://github.com/veler/DevToys/assets/7417301/9b437ffe-1c11-421e-9f6d-94a4284996d8


Both of your suggestions (ContentControl Alignment or child of Border) allow the DataGrid to consume as much vertical space as it wants, causing the parent tool page to scroll rather than the DataGrid itself.

P.S. Thank you so much for working with me on this. I think it's a really useful feature and I hope we can do it in a way that works and is nicely written. 😊

Reply:
hello,
Thanks for the explanation. That's fair then! :D Let's keep this solution. I'm just super surprised of the need of this `RunIdleAsync ` since I never needed it before 😅

Reply:
Sounds good, and trust me, I would love to have found a less hacky solution! The real problem is that we need the help of the `CodeEditor` to size the `DataGrid` when the `Border` doesn't work. `RunIdleAsync` is just the only way I could make the `CodeEditor` size after being made visible.

Reply:
That's fair 😁
I'm hoping we won't need anything hacky in DevToys 2.0 (made with Blazor, not Xaml) 🤞

Reply:
Sounds good. I'm a huge fan of XAML despite its quirks. I hope Blazor holds up! I might have to learn it. 😊

Reply:
I'm a huge Xaml fan too! Been doing it since 2006. For the first time, I meet a case where Xaml isn't the right answer. We can thank our dependency on a efficient and scalable text editor  along with cross-platform need. 😅